econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -362,7 +549,7 @@ def fit(self, X, y, warm_start=False):
                               "since the previous call to fit")
                 warm_start = False
 
-            # TODO: bail out also if categorical columns, classification changed?
+            # TODO: bail out also if categorical columns, classification, random_state changed?
```

Comment:
I suspect we should not retrain a model_y when only heterogeneity_model has changed. 

If that is the case, then why are we also not checking if "heterogeneity_inds has changed here.

Reply:
Not sure I understand the question; however I have now changed the logic so that we do not retrain a y model when heterogeneity_model changes

Reply:
But maybe a partial answer to your question is that even if some heterogeneity_inds have changed (so that models need to be retrained there), others may not have and so we want to keep most of the warm_start logic intact.